### PR TITLE
Feature/non user students

### LINKED ISSUE #4: Student Feature
Create the student entity. Show more details about students in the app. Students don't need to be authenticated users initially.

### PR DESCRIPTION
**Pull request by arjadas**
This PR introduces the Students feature across the full stack, including backend, API layer, and frontend integration.

**Scope of Work**

**Backend**

- Extended DB contract and implementation to support Students
- Added DTOs and request/response contracts for:
  - List
  - Detail
  - Create
  - Update
- Implemented query and command handlers
- Added full CRUD operations with role-based access control (RBAC)

**Frontend**

- Implemented API client for Students endpoints
- Added reusable hooks for CRUD operations
- Built UI components for:
  - Student list
  - Student detail
- Wired teacher-facing screens

**Routing**

- Integrated new routes for Students feature
- Connected list and detail views to navigation

**Key Features**

- End-to-end student management (list, view, create, update)
- RBAC enforcement at the handler level
- Consistent contract between backend and frontend
- Modular hooks for easier data fetching and mutations

**Notes**

- This PR delivers the full vertical slice of the Students feature in one pass
- Follow-up improvements can include:
  - Additional test coverage (especially RBAC scenarios)
  - UI refinements and edge-case handling

### CODE REVIEW

File: frontend/app/(app)/(teacher)/students/[studentId]/index.tsx [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    }
+
+    const updatePayload: UpdateStudentRequest = {
+      firstName: payload.firstName,
+      lastName: payload.lastName,
+      email: payload.email,
+      phone: payload.phone,
+      notes: payload.notes,
+      metadata: payload.metadata,
+    };
```
[the remark this answers is not in the document]
Comment (arjadas) [reply frontend/app/(app)/(teacher)/students/[studentId]/index.tsx:67]: I believe the system is working as intended. 

We intentionally treat null as “no change” and only apply updates when a value is provided. This makes the command behave like a partial update (PATCH-style) and prevents accidental overwrites. Empty string is used when a field should be explicitly cleared.

File: backend/src/TuitionIQ.Application/Features/Students/Queries/GetStudentsQuery.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      studentsQuery = studentsQuery.Where(student => student.Status == parsedStatus);
+    }
+
+    if (!string.IsNullOrWhiteSpace(request.Search))
+    {
+      var normalizedSearch = request.Search.Trim().ToLower();
+      studentsQuery = studentsQuery.Where(student =>
+        student.FirstName.ToLower().Contains(normalizedSearch)
+        || student.LastName.ToLower().Contains(normalizedSearch));
+    }
```
[the remark this answers is not in the document]
Comment (arjadas) [reply backend/src/TuitionIQ.Application/Features/Students/Queries/GetStudentsQuery.cs:88]: used PostgreSQL EF.Functions.ILike

File: backend/src/TuitionIQ.Application/Features/Students/Queries/GetStudentsQuery.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    var callerRole = await _dbContext.FirstOrDefaultAsync(membershipRoleQuery, cancellationToken);
+    if (callerRole is null)
+    {
+      throw new ForbiddenException("You are not allowed to view students in this organization.");
+    }
+
+    IQueryable<Student> studentsQuery = _dbContext.Students
+      .Where(student => student.OrganizationId == request.OrganizationId);
+
```
[the remark this answers is not in the document]
Comment (arjadas) [reply backend/src/TuitionIQ.Application/Features/Students/Queries/GetStudentsQuery.cs:68]: right now, there's only teachers who are the owners so we will look into this RBAC issue later.
